osu.Game/Online/OnlineViewContainer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.API;
+using osu.Game.Online.Placeholders;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A <see cref="Container"/> for dislaying online content who require a local user to be logged in.
+    /// Shows its children only when the local user is logged in and supports displaying a placeholder if not.
+    /// </summary>
+    public class OnlineViewContainer : Container, IOnlineComponent
+    {
+        private readonly Container content;
+        private readonly Container placeholderContainer;
+        private readonly Placeholder placeholder;
+
+        private const int transform_time = 300;
+
+        protected override Container<Drawable> Content => content;
+
+        public OnlineViewContainer(string placeholder_message)
+        {
+            InternalChildren = new Drawable[]
+            {
+                content = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                placeholderContainer = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Alpha = 0,
+                    Child = placeholder = new LoginPlaceholder(placeholder_message)
+                },
+            };
+        }
+
+        public void APIStateChanged(IAPIProvider api, APIState state)
```

Comment:
Should be `virtual` to allow components that inherit this and also want `IOnlineComponent` features to override this.